security/pkg/nodeagent/vm/citadelagent_v2.go

```diff
@@ -159,6 +168,10 @@ func (na *citadelAgent) sendCSRUsingCANewProtocol() ([]byte, []string, error) {
 
 // getProjectID returns the project id from the provided jwt or an error.
 func getProjectID(jwt string) (string, error) {
+	// For unit tests
```

Comment:
no need for special implementation here? right? how about a static jwt for testing purpose.you're just doing a decoding and extraction, no check on expiration time.

Reply:
and you get more test coverage by walking through the implementation below.

Reply:
That's a good idea.